src/frontend/apps/impress/src/core/AppProvider.tsx

```diff
@@ -13,12 +17,16 @@ import { ConfigProvider } from './config/';
  * QueryClient:
  *  - defaultOptions:
  *    - staleTime:
- *      - global cache duration - we decided 3 minutes
- *      - It can be overridden to each query
+ *      - global time until cache is considered stale and will be refetched in the background
+ *        - instant if debug flag "no-cache" active - 3 minutes otherwise
+ *    - gcTime:
+ *      - global time until cache is purged from the persister and needs to be renewed
+ *        - since its cached in localStorage, we can set it to a long time (48h)
  */
 const defaultOptions = {
   queries: {
-    staleTime: 1000 * 60 * 3,
+    staleTime: debug.enabled('no-cache') ? 0 : 1000 * 60 * 3, // 3 minutes
+    gcTime: debug.enabled('no-cache') ? 0 : 1000 * 60 * 60 * 48, // 48 hours
```

Comment:
About the debug, I am not sure it is the best way to do. 
As a developer I like to be near the production setting, it helps me in this case to see which query I have to invalidate. If I have `no-cache`, I will not know that when I change the **title of a document** I need to invalidate the cache of the **document list** request.

After if you don't want cache, we could set these variables from settings:
```suggestion
    staleTime: process.env.NEXT_PUBLIC_CACHE_STALE_TIME,
    gcTime: process.env.NEXT_PUBLIC_CACHE_GC_TIME
```
By default they will be in `.env`:
```
NEXT_PUBLIC_CACHE_STALE_TIME=1000 * 60 * 3
NEXT_PUBLIC_CACHE_GC_TIME=1000 * 60 * 60 * 48
```
but in `.env.local` you could set them as you want, `.env.local` is not versioned:
```
NEXT_PUBLIC_CACHE_STALE_TIME=0
NEXT_PUBLIC_CACHE_GC_TIME=0
```

Reply:
Thanks for the feedback. 

1.  **Debug Mode and Cache:** The `debug` flag for cache control is off by default, aligning with production behavior. Developers who enable it for local debugging are intentionally altering cache settings for temporary diagnostic purposes.

2.  **Data Updates and Cache Invalidation:** For data modifications (e.g., updating a document title), we should adhere to the mutation pattern ([as demonstrated here](https://github.com/suitenumerique/docs/pull/857#discussion_r2098098188)). This ensures that related queries are correctly invalidated and fresh data is fetched.

3.  **Configurability of `staleTime` and `gcTime`:**
    * Making the default `staleTime` and `gcTime` configurable via environment variables could introduce complexity. Because each `useQuery` can override defaults - we would need to add env configuration for each `useQuery` that has a different `staleTime` and `gcTime`.
    * Generally, we, as developers designing the queries, are best positioned to determine appropriate `staleTime` and `gcTime` values based on the nature of the data. Allowing arbitrary instance-level customization or overly flexible environment configurations might lead to inconsistent caching behavior and unexpected issues. It's preferable to establish sensible, well-reasoned defaults within the codebase for different types of queries.